x-pack/solutions/observability/packages/utils_server/es/storage/index_adapter/index.ts

```diff
@@ -68,6 +65,20 @@ function catchConflictError(error: Error) {
   throw error;
 }
 
+/*
+ * When calling into Elasticsearch, the stack trace is lost.
+ * If we create an error before calling, and append it to
+ * any stack of the caught error, we get a more useful stack
+ * trace.
+ */
+function wrapEsCall<T>(p: Promise<T>): Promise<T> {
+  const error = new Error();
```

Comment:
any concern with generating a stacktrace on every es call or is the trace lazily generated when accessing `.stack` ? Not sure how big of a performance hit but there must be a cost, is the instrumentation going to capture that ?

Reply:
per https://developer.chrome.com/blog/faster-stack-traces

> That project was a great success and drastically reduced the number of issues related to stack trace collection. Most of the information provided via the non-standard error.stack property had also been computed lazily and only when it was really needed, but as part of the refactoring we applied the same trick to v8::StackFrame objects. **All information about the stack frame is computed the first time any method was invoked on it.**

Reply:
(the other overhead attached to calling out to ES, such as serializing and deserializing is probably significantly bigger anyway)